packages/client/src/devices/InputMediaDeviceManager.ts

```diff
@@ -191,16 +172,32 @@ export abstract class InputMediaDeviceManager<
     stopTracks ? this.stopTrack() : this.muteTrack();
   }
 
+  protected async muteStream(stopTracks: boolean = true) {
+    if (!this.state.mediaStream) {
+      return;
+    }
+    this.logger('debug', `${stopTracks ? 'Stopping' : 'Disabling'} stream`);
+    if (this.call.state.callingState === CallingState.JOINED) {
+      await this.stopPublishStream(stopTracks);
+    }
+    this.muteLocalStream(stopTracks);
+    if (this.getTrack()?.readyState === 'ended') {
+      // @ts-expect-error release() is present in react-native-webrtc and must be called to dispose the stream
+      if (typeof this.state.mediaStream.release === 'function') {
+        // @ts-expect-error
+        this.state.mediaStream.release();
+      }
+      this.state.setMediaStream(undefined);
+    }
+  }
+
   protected async unmuteStream() {
     this.logger('debug', 'Starting stream');
     let stream: MediaStream;
     if (this.state.mediaStream && this.getTrack()?.readyState === 'live') {
       stream = this.state.mediaStream;
       this.unmuteTrack();
     } else {
-      if (this.state.mediaStream) {
```

Comment:
As discussed with @szuperaz, this is not at all called in any case of the `unmuteStream` execution. Hence, it is removed.